apps/homescreen/js/app.js

```diff
@@ -364,6 +357,46 @@ const SETTINGS_VERSION = 0;
 
       this.small = settings.small || false;
       this.icons.classList.toggle('small', this.small);
+
+      // Monitor global homescreen settings
+      if (!navigator.getDataStores) {
+        console.error('Datastore API unavailable');
+        return;
+      }
+
+      navigator.getDataStores(SETTINGS_STORE).then(stores => {
+        if (stores.length < 1) {
+          return Promise.reject('Settings datastore inaccessible');
+        }
+
+        var syncSmallSetting = () => {
+          stores[0].get(COLUMNS_SETTING).then(cols => {
+            this.toggleSmall(cols > 3);
+          });
+        };
+
+        stores[0].addEventListener('change', e => {
+          if (e.id === COLUMNS_SETTING) {
+            syncSmallSetting();
+          }
+        });
+        syncSmallSetting();
+      }, e => {
+        console.error('Error retrieving home screen settings datastore:', e);
+      });
+    },
```

Comment:
This seems to harm readability. I know the logic is simple but maybe you can move this section to its own script (like datastore.js and pagesstore.js).